test/Ocean/Hydrostatic_Boussinesq/test_divergence_free.jl

```diff
@@ -0,0 +1,203 @@
+using MPI
+using CLIMA
+using CLIMA.Mesh.Topologies
+using CLIMA.Mesh.Grids
+using CLIMA.DGmethods
+using CLIMA.DGmethods.NumericalFluxes
+using CLIMA.MPIStateArrays
+using CLIMA.LowStorageRungeKuttaMethod
+using CLIMA.ODESolvers
+using CLIMA.GenericCallbacks
+using CLIMA.VariableTemplates: flattenednames
+using CLIMA.HydrostaticBoussinesq
+using LinearAlgebra
+using StaticArrays
+using Logging, Printf, Dates
+using CLIMA.VTK
+using CLIMA.PlanetParameters: grav
+import CLIMA.HydrostaticBoussinesq: ocean_init_aux!, ocean_init_state!,
+                                    Coastline, OceanFloor, OceanSurface,
+                                    ocean_boundary_state!
+import CLIMA.DGmethods: update_aux!, vars_state, vars_aux
+using CLIMA.VariableTemplates
+using Test
+
+@static if haspkg("CuArrays")
+  using CUDAdrv
+  using CUDAnative
+  using CuArrays
+  CuArrays.allowscalar(false)
+end
+
+HBModel   = HydrostaticBoussinesqModel
+HBProblem = HydrostaticBoussinesqProblem
+
+@inline function ocean_boundary_state!(m::HBModel, bctype, x...)
+  if bctype == 1
+    ocean_boundary_state!(m, Coastline(), x...)
+  elseif bctype == 2
+    ocean_boundary_state!(m, OceanFloor(), x...)
+  elseif bctype == 3
+    ocean_boundary_state!(m, OceanSurface(), x...)
+  end
+end
+
+struct HomogeneousSimpleBox{T} <: HydrostaticBoussinesqProblem
+  Lˣ::T
+  Lʸ::T
+  H::T
+  τₒ::T
+  fₒ::T
+  β::T
+end
+
+HSBox = HomogeneousSimpleBox
+
+# aux is Filled afer the state
+function ocean_init_aux!(::HBModel, P::HSBox, A, geom)
+  FT = eltype(A)
+  @inbounds y = geom.coord[2]
+
+  Lʸ = P.Lʸ
+  τₒ = P.τₒ
+  fₒ = P.fₒ
+  β  = P.β
+
+  A.τ  = -τₒ * cos(y * 2π / Lʸ)
+  A.f  =  fₒ + β * y
+end
+
+function ocean_init_state!(p::HSBox, state, aux, coords, t)
+  @inbounds z = coords[3]
+  @inbounds H = p.H
+
+  state.u = @SVector [rand(),rand()]
```

Comment:
@jkozdon , I get an "unsupported call through a literal pointer" error from this line when running on the GPU. 

I tried to get around it by doing something like `u = rand(); v = rand()` and then passing `u` and `v` through the state vector, but still got the error.

Any suggestions? Besides starting with [0,0] of course.

Reply:
See [comment below](https://github.com/climate-machine/CLIMA/pull/535/files/246fe87ca6c9489e33f6c04775fa62f359a0a4ef#r348190197) on CPU based initialization